<?xml version="1.0"?>
<launch>
  <arg name="img" default="/image_raw"/>
  <node name="class_tsn_rgb" pkg="caffe_tsn_ros" type="class_tsn_OS.py" respawn="false" output="screen" >
    <!-- <node name="class_tsn_rgb" pkg="caffe_tsn_ros" type="class_tsn_rgb3.py" respawn="false" output="screen" ns="subject"> -->

    <remap from="video_topic" to="$(arg img)"/>
	  <param name="device_id" value="0"/>
    <param name="dataset" value="hmdb51"/>
    <param name="split" value="1"/>
    <!-- TODO: frame_size should maybe be variable and maybe I should publish to
this node only a box around the subject. this would be a big change and would
require retraining and even maybe another node/cnn thingy publishing info about
environment and objects and an additional network to integrate it all (and get
back the accuracy we lose by not taking into account meaningful environment and
objects)  -->
    <param name="framesize_width" value="340"/>
    <param name="framesize_height" value="256"/>

    <rosparam param="action_list">['brush_hair','cartwheel','catch','chew','clap',
      'climb','climb_stairs','dive','draw_sword','dribble','drink','eat',
      'fall_floor','fencing','flic_flac','golf','handstand','hit','hug','jump',
      'kick','kick_ball','kiss','laugh','pick','pour','pullup','punch','push',
      'pushup','ride_bike','ride_horse','run','shake_hands','shoot_ball',
      'shoot_bow','shoot_gun','sit','situp','smile','smoke','somersault',
      'stand','swing_baseball','sword','sword_exercise','talk','throw','turn',
      'walk','wave']
    </rosparam>
   <rosparam param="choose_list">['brush_hair','chew','clap',
     'drink','eat',
     'fall_floor','jump',
     'laugh','pick','pour',
     'run',
     'sit','smile',
     'stand','talk','turn',
     'walk','wave']
    </rosparam>
    <!-- <rosparam param="choose_list">['brush_hair','cartwheel','catch','chew','clap',
       'climb','climb_stairs','dive','draw_sword','dribble','drink','eat',
       'fall_floor','fencing','flic_flac','golf','handstand','hit','hug','jump',
       'kick','kick_ball','kiss','laugh','pick','pour','pullup','punch','push',
       'pushup','ride_bike','ride_horse','run','shake_hands','shoot_ball',
       'shoot_bow','shoot_gun','sit','situp','smile','smoke','somersault',
       'stand','swing_baseball','sword','sword_exercise','talk','throw','turn',
       'walk','wave']
     </rosparam> -->
    <!-- <param name="choose_list" value="['dive','jump','hug']"/> -->
	  <param name="classification_frame_window" value="50"/>
  </node>
</launch>
